src/main/java/appeng/fluids/storage/FluidCellInventory.java

```diff
@@ -0,0 +1,270 @@
+		{
+			return null;
+		}
+	}
+
+	public static boolean isCell( final ItemStack i )
+	{
+		if( i == null )
+		{
+			return false;
+		}
+
+		final Item type = i.getItem();
+		if( type instanceof IStorageCell )
+		{
+			if( ( (IStorageCell) type ).getChannel() == AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class ) )
+			{
+				return ( (IStorageCell) type ).isStorageCell( i );
+			}
+		}
+
+		return false;
+	}
+
+	@Override
+	public IAEFluidStack injectItems( final IAEFluidStack input, final Actionable mode, final IActionSource src )
+	{
+		if( input == null )
+		{
+			return null;
+		}
+		if( input.getStackSize() == 0 )
+		{
+			return null;
+		}
+
+		if( this.cellType.isBlackListed( this.i, input ) )
+		{
+			return input;
+		}
+
+		final FluidStack sharedFluidStack = input.getFluidStack();
+
+		final IAEFluidStack l = this.getCellItems().findPrecise( input );
+		if( l != null )
+		{
+			final long remainingItemSlots = this.getRemainingItemCount();
+			if( remainingItemSlots < 0 )
+			{
+				return input;
+			}
+
+			if( input.getStackSize() > remainingItemSlots )
+			{
+				final IAEFluidStack r = input.copy();
+				r.setStackSize( r.getStackSize() - remainingItemSlots );
+				if( mode == Actionable.MODULATE )
+				{
+					l.setStackSize( l.getStackSize() + remainingItemSlots );
+					this.updateItemCount( remainingItemSlots );
+					this.saveChanges();
+				}
+				return r;
+			}
+			else
+			{
+				if( mode == Actionable.MODULATE )
+				{
+					l.setStackSize( l.getStackSize() + input.getStackSize() );
+					this.updateItemCount( input.getStackSize() );
+					this.saveChanges();
+				}
+				return null;
+			}
+		}
+
+		if( this.canHoldNewItem() ) // room for new type, and for at least one item!
+		{
+			final int remainingItemCount = (int) this.getRemainingItemCount() - this.getBytesPerType() * itemsPerByte;
+			if( remainingItemCount > 0 )
+			{
+				if( input.getStackSize() > remainingItemCount )
+				{
+					final FluidStack toReturn = sharedFluidStack.copy();
+					toReturn.amount = sharedFluidStack.amount - remainingItemCount;
+					if( mode == Actionable.MODULATE )
+					{
+						final FluidStack toWrite = sharedFluidStack.copy();
+						toWrite.amount = remainingItemCount;
+
+						this.cellItems.add( AEFluidStack.fromFluidStack( toWrite ) );
+						this.updateItemCount( toWrite.amount );
+
+						this.saveChanges();
+					}
+					return AEFluidStack.fromFluidStack( toReturn );
+				}
+
+				if( mode == Actionable.MODULATE )
+				{
+					this.updateItemCount( input.getStackSize() );
+					this.cellItems.add( input );
+					this.saveChanges();
+				}
+
+				return null;
+			}
+		}
+
+		return input;
+	}
+
+	@Override
+	public IAEFluidStack extractItems( final IAEFluidStack request, final Actionable mode, final IActionSource src )
+	{
+		if( request == null )
+		{
+			return null;
+		}
+
+		final long size = Math.min( Integer.MAX_VALUE, request.getStackSize() );
+
+		IAEFluidStack Results = null;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS00117)